packages/flutter/test/material/carousel_test.dart

```diff
@@ -1193,6 +1193,55 @@ void main() {
     // No exception.
     expect(tester.takeException(), isNull);
   });
+
+  testWidgets('The shrinkExtent should keep the same when the item is tapped', (WidgetTester tester) async {
+    int offset = 0;
+
+    await tester.pumpWidget(
+      StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return MaterialApp(
+            home: Scaffold(
+              body: Center(
+                child: ConstrainedBox(
+                  constraints: const BoxConstraints(maxHeight: 200),
+                  child: CarouselView(
+                    itemExtent: 330,
+                    onTap: (int idx) => setState(() {
+                      offset = idx;
+                    }),
+                    children: List<Widget>.generate(20, (int index) {
+                      index += offset;
+                      return ColoredBox(
+                        color: Colors.primaries[index % Colors.primaries.length].withOpacity(0.5),
+                        child: Center(
+                          child: Text('Item $index'),
+                        ),
+                      );
+                    }),
+                  ),
+                ),
+              ),
+            ),
+          );
+        }
+      )
+    );
+
+    await tester.pumpAndSettle();
+
+    expect(tester.getRect(getItem(0)).width, 330.0);
+
+    final Finder item1 = find.text('Item 1');
+    await tester.tap(find.ancestor(of: item1, matching: find.byType(Stack)));
+
+    await tester.pumpAndSettle();
+
+    expect(tester.getRect(getItem(1)).width, 330.0);
+    expect(tester.getRect(getItem(2)).width, 330.0);
+    // This should be less than 330.0 because the item is shrank.
+    expect(tester.getRect(getItem(3)).width, 140.0);
```

Comment:
In the original fix, when we tap a carousel item 1, the item index is changed based on the offset because of the code `index += offset;` which make the carousel item index look a little odd and item 0 disappear on screen. I think we don't need to "reset" the index in this test, so on screen, there's always item 0, item 1 and 2:)
```suggestion
    expect(tester.getRect(getItem(0)).width, 330.0);
    expect(tester.getRect(getItem(1)).width, 330.0);
    // This should be less than 330.0 because the item is shrank; width is 800.0 - 330.0 - 330.0
    expect(tester.getRect(getItem(2)).width, 140.0);
```